ios/FluentUI/Bottom Commanding/CommandingItem.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// An object representing a command.
+///
+/// `CommandingItem` defines the high level properties and behavior of a command. Its visual representation is determined by
+/// the `BottomCommandingController`.
+@objc(MSFCommandingItem)
+open class CommandingItem: NSObject {
+
+    /// A closure that's called when the command is triggered
+    @objc open var action: (CommandingItem) -> Void
+
+    /// The title of the command item.
+    @objc open var title: String {
+        didSet {
+            if title != oldValue {
+                delegate?.commandingItem(self, didChangeTitleTo: title)
+            }
+        }
+    }
+
+    /// A `UIImage` to be displayed with the command.
+    @objc open var image: UIImage {
+        didSet {
+            if image != oldValue {
+                delegate?.commandingItem(self, didChangeImageTo: image)
+            }
+        }
+    }
+
+    /// A `UIImage` used when the command is represented as a button in selected state.
+    @objc open var selectedImage: UIImage? {
+        didSet {
+            if selectedImage != oldValue {
+                delegate?.commandingItem(self, didChangeSelectedImageTo: selectedImage)
+            }
+        }
+    }
+
+    /// Used in large content viewer if this command is represented using a view that cannot scale with Dynamic Type.
+    ///
+    /// When this is `nil`, `image` will be used instead.
+    @objc open var largeImage: UIImage? {
+        didSet {
+            if largeImage != oldValue {
+                delegate?.commandingItem(self, didChangeLargeImageTo: largeImage)
+            }
+        }
+    }
+
+    /// Indicates whether the command is currently on.
+    ///
+    /// When `isToggleable` is `true`, this property is toggled automatically before `action` is called.
+    @objc open var isOn: Bool {
```

Comment:
should it be isSelected? or is it for booleancell only?

Reply:
I felt like isOn is a better general name that covers both the boolean cell and regular button cases. isSelected really only makes sense for buttons.

Reply:
Thought about this more, and "selected" is a name that is also used for checkboxes, and for design tokens (button-selected vs button-rest, etc). Also I had to do a double take to think "Does isOn actually mean isEnabled?" 
I'm feeling isSelected more now..

Reply:
I'll point out that this was already reviewed and approved 😄 I guess I'm down to change it, but the reasoning here was that isOn is more view-agnostic. It just says the command is on, regardless of how it's represented. Switches use isOn, buttons use isSelected. isSelected doesn't make as much sense with switches. Another issue is that cell with a switch in it technically has independent isSelected and isOn states. We don't expose that through this API right now, but using isSelected would cause issues in that area in the future, for example if we wanted to support grouped selectable cells with checkboxes in them.